### DISCUSSION TITLE
Sanjay Jaiswal - GitHub Externship Proposal

### DISCUSSION THREAD

**Discussion by sanjay-xdr** [hoppscotch/hoppscotch#1633] [Q&A]
**Applicationnumber:** 805_San86_tfa_258

**Full name:** Sanjay Jaiswal

**Email:** sanjayjaiswal586@gmail.com

**GitHub username:** sanjay-xdr

**Prior experience:**
- [x] HTML
- [x] CSS
- [x] SCSS
- [ ] Tailwind CSS
- [x] JavaScript
- [ ] TypeScript
- [ ] Vue.js
- [ ] Nuxt.js
- [ ] NestJS
- [ ] Firebase
- [ ] Prisma
- [ ] GraphQL
- [ ] PostgreSQL
- [ ] Go lang

---

**Proposal type:**
- [x] Existing feature request
- [ ] New feature proposal 
- [x] Refactor existing feature

**Proposal title:** Improving the hoppscotch extension and Improving the UI

**Proposal description:**  
- Making the hoppscotch extension to run the api test on the same window and adding the basic features e.g. passing the parameters and response on the same window for the ease of  the users
- Adding a full screen feature to hoppscotch website and displaying the useful features e.g. parameters, header, authentication and response on the full screen but at the same time other features will be working like maintaining the users request history behind the scene

**Proposal roadmap:** 
1. Improving the extension adding the test api feature on the same window and designing the UI as well using HTML,CSS and JavaScript
2. Creating the full screen feature on main website and displaying the important features

**Comment by AndrewBastin** [comment]
````
Improving the extension adding the test api feature on the same window
````

Can you explain to me what you mean by the `same window' ?

**Reply by sanjay-xdr** [reply]
Right now all the chrome extensions that are present on chrome store to test an api ,there working is ,when you click on the extension it opens up a new  tab for api testing. But same window here refers to the popup that opens up when u click on an extension and on the same popup window developer can test his/her api.
